### PR TITLE
operatingsystems-linux-snmp: Adding Disk IO service discovery

### PR DESCRIPTION
**Pull request by sdepassio**
## Target version (i.e. version that this PR changes)

- [ ] 21.10.x (staging)
- [ ] 22.04.x (staging)
- [ ] 22.10.x (staging)
- [ ] 23.04.x (staging)
- [ ] Cloud (staging)
- [x] Monitoring Connectors (staging)
- [ ] 23.10.x (next)

### CODE REVIEW

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 1]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
-| Obligatoire | Nom              | Description                                    |
-| :---------- | :--------------- | :--------------------------------------------- |
-|             | SNMPEXTRAOPTIONS | Configure your own SNMPv3 credentials combo    |
+</TabItem>
+<TabItem value="Disk-Generic-Id" label="Disk-Generic-Id">
+
+| Macro        | Description                                                                                         | Valeur par défaut                             | Obligatoire |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKID       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:421]: description

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 2]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
+| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). Toutes les options sont listées [ici](#options-disponibles) | --filter-perfdata='storage.space\|used\|free' |             |
+
+</TabItem>
+<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">
 
-## FAQ
+| Macro        | Description                                                                                         | Valeur par défaut                             | Obligatoire |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKNAME     | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:433]: description

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 3]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
 
-### Comment tester un contrôle en ligne de commande et que signifient les options principales ?
+</TabItem>
+<TabItem value="Disk-Global" label="Disk-Global">
 
-A partir du moment ou le Plugin est installé, vous pouvez tester directement celui-ci en ligne de commande depuis votre collecteur Centreon avec l'utilisateur *centreon-engine*:
+| Macro        | Description                                                                                         | Valeur par défaut                                       | Obligatoire |
+|:-------------|:----------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|
+| FILTER       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    | .*                                                      |             |
+| TRANSFORMSRC |                                                                                                     |                                                         |             |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:445]: description

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 4]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
-    --plugin=os::linux::snmp::plugin \
-    --mode=cpu \
-    --help
+/usr/lib/centreon/plugins/centreon_linux_snmp.pl \
+	--plugin=os::linux::snmp::plugin \
+	--list-mode
 ```
 
-### UNKNOWN: SNMP GET Request : Timeout
+Le plugin apporte les modes suivants :
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:686]: rubriques de troubleshooting?

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md (hunk 1 above)
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:421]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                               |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 5]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
-| :---------- | :--------------- | :--------------------------------------------- |
-|             | SNMPEXTRAOPTIONS | Configure your own SNMPv3 credentials combo    |
+</TabItem>
+<TabItem value="Disk-Generic-Id" label="Disk-Generic-Id">
+
+| Macro        | Description                                                                                         | Valeur par défaut                             | Obligatoire |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKID       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
+| TRANSFORMDST |                                                                                                     |                                               |             |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:422]:
```suggestion
| TRANSFORMDST | Modify the disk name displayed by using a regular expression (pattern substitution). |                                               |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 6]
```diff
@@ -152,95 +283,823 @@ The SNMP server must be restarted each time the configuration is modified. Also
...
+| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --filter-perfdata='storage.space\|used\|free' |             |
+
+</TabItem>
+<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">
+
+| Macro        | Description                                                                                         | Default value                                 | Mandatory   |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKNAME     | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
+| TRANSFORMDST |                                                                                                     |                                               |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:433]:
```suggestion
| TRANSFORMDST | Modify the disk name displayed by using a regular expression (pattern substitution). |                                               |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 7]
```diff
@@ -152,95 +283,823 @@ The SNMP server must be restarted each time the configuration is modified. Also
...
+
+</TabItem>
+<TabItem value="Disk-Global" label="Disk-Global">
+
+| Macro        | Description                                                                                         | Default value                                           | Mandatory   |
+|:-------------|:----------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|
+| FILTER       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    | .*                                                      |             |
+| TRANSFORMSRC |                                                                                                     |                                                         |             |
+| TRANSFORMDST |                                                                                                     |                                                         |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:445]:
```suggestion
| TRANSFORMDST | Modify the disk name displayed by using a regular expression (pattern substitution). |                                                         |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 8]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
+<TabItem value="Disk-Global" label="Disk-Global">
 
-A partir du moment ou le Plugin est installé, vous pouvez tester directement celui-ci en ligne de commande depuis votre collecteur Centreon avec l'utilisateur *centreon-engine*:
+| Macro        | Description                                                                                         | Valeur par défaut                                       | Obligatoire |
+|:-------------|:----------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|
+| FILTER       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    | .*                                                      |             |
+| TRANSFORMSRC |                                                                                                     |                                                         |             |
+| TRANSFORMDST |                                                                                                     |                                                         |             |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:446]:
```suggestion
| TRANSFORMDST | Modify the disk name displayed by using a regular expression (pattern substitution). |                                                         |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 9]
```diff
@@ -153,93 +286,823 @@ Assurez vous que le processus SNMP est configuré pour démarrer automatiquement
...
+
+</TabItem>
+<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">
 
-## FAQ
+| Macro        | Description                                                                                         | Valeur par défaut                             | Obligatoire |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKNAME     | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
+| TRANSFORMDST |                                                                                                     |                                               |             |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:434]:
```suggestion
| TRANSFORMDST | Modify the disk name displayed by using a regular expression (pattern substitution). |                                               |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md (hunk 2 above)
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:433]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                               |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md (hunk 3 above)
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:445]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                                         |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 10]
```diff
@@ -152,95 +283,823 @@ The SNMP server must be restarted each time the configuration is modified. Also
...
+| CRITICALIDLE | Critical threshold in percent                                                                       | 10:               |             |
+| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --verbose         |             |
+
+</TabItem>
+<TabItem value="Disk-Generic-Id" label="Disk-Generic-Id">
+
+| Macro        | Description                                                                                         | Default value                                 | Mandatory   |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKID       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:420]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                               |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 11]
```diff
@@ -152,95 +283,823 @@ The SNMP server must be restarted each time the configuration is modified. Also
...
+| TRANSFORMDST |                                                                                                     |                                               |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:421]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                               |             |
| TRANSFORMDST | Modify the disk name displayed by using a regular expression (pattern substitution). |                                               |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 12]
```diff
@@ -152,95 +283,823 @@ The SNMP server must be restarted each time the configuration is modified. Also
...
+| CRITICAL     | Critical threshold                                                                                  | 90                                            |             |
+| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --filter-perfdata='storage.space\|used\|free' |             |
+
+</TabItem>
+<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">
+
+| Macro        | Description                                                                                         | Default value                                 | Mandatory   |
+|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
+| DISKNAME     | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |
+| TRANSFORMSRC |                                                                                                     |                                               |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:432]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                               |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 13]
```diff
@@ -152,95 +283,823 @@ The SNMP server must be restarted each time the configuration is modified. Also
...
+| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --filter-perfdata='storage.space\|used\|free' |             |
+
+</TabItem>
+<TabItem value="Disk-Global" label="Disk-Global">
+
+| Macro        | Description                                                                                         | Default value                                           | Mandatory   |
+|:-------------|:----------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|
+| FILTER       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    | .*                                                      |             |
+| TRANSFORMSRC |                                                                                                     |                                                         |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:444]:
```suggestion
| TRANSFORMSRC | Modify the disk name displayed by using a regular expression (pattern matching). |                                                         |             |
```

File: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md [hunk 14]
```diff
@@ -5,121 +5,252 @@ title: Linux SNMP
...
 
-<Tabs groupId="sync">
-<TabItem value="cpu" label="cpu">
+> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.
 
-| Metric name                        | Description                                   |
-| :--------------------------------- | :-------------------------------------------- |
-| cpu.utilization.percentage         | CPU utilization. Unit : %                     |
-| core.cpu.utilization.percentage    | Per Core CPU utilization. Unit : %            |
+</TabItem>
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp.md:157]:
```suggestion
To work properly, the Inodes mode needs the following directive to be set in the SNMP server configuration file:

```includeAllDisks 10%```

</TabItem>
```

**Review by omercier — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**